### DISCUSSION TITLE
Standalone nixvim produces strange behavior

### DISCUSSION THREAD

**Discussion by djacu** [nix-community/nixvim#1575] [Q&A]
Hey all

I tried following the standalone usage docs so I can have a system build pinned to 23.11 but use the latest tip of nixvim, and I'm running into some strange behavior.

This is the module I setup based the standalone usage docs.

```
{ inputs, ... }:
let
  config =
    { ... }:
    {
      colorschemes.gruvbox.enable = true;
      plugins.lightline.enable = true;
    };
  nixvim' = inputs.nixvim.legacyPackages.x86_64-linux;
  nixvimModule = {
    pkgs = inputs.nixpkgs-unstable.legacyPackages.x86_64-linux;
    module = config;
  };
in
{
  environment.systemPackages = [ (nixvim'.makeNixvimWithModule nixvimModule) ];
}
```

It gives me the classic `error: attribute 'base16-nvim' missing` error that you see when trying to mix nixos 23.11 with nixvim unstable.

If I move the same code to the top level flake as a package like so, it works fine

```
packages.x86_64-linux.neovim =
        let
          config =
            { ... }:
            {
              colorschemes.gruvbox.enable = true;
              plugins.lightline.enable = true;
            };
          nixvim' = inputs.nixvim.legacyPackages.x86_64-linux;
          nixvimModule = {
            pkgs = inputs.nixpkgs-unstable.legacyPackages.x86_64-linux;
            module = config;
          };
        in
        nixvim'.makeNixvimWithModule nixvimModule;
```

With the caveat that I get a `lightline.vim: Could not load colorscheme gruvbox` error. But at least it runs.

So I'm trying to understand if and where I went wrong.

**Comment by MattSturgeon** [comment]
Has your `nixpkgs-unstable` input need updated recently in your `flake.lock`? Using `nix flake update` or `nix flake lock --update-input nixpkgs-unstable`

If not it could be out of date enough to cause issues 🤔

**Reply by djacu** [reply]
It was a freshly created project at the time of posting so the flake inputs should be pretty fresh.

**Comment by fluxoz** [comment]
I am currently having the same issue:

![image](https://github.com/user-attachments/assets/7e3e8777-99eb-4188-acd9-da2d1c017c5d)


My home-manager config:
```
    programs.nixvim = {
	enable = true;
        defaultEditor = true;
        extraPlugins = with pkgs.vimPlugins; [
        	gruvbox-material
        ];
	colorschemes.gruvbox.enable = true;
        plugins.lightline.enable = true;
    };
```
My flake.nix:
```
nixvim = {
	url = "github:nix-community/nixvim/nixos-24.05";
	inputs.nixpkgs.follows = "nixpkgs";
    };
```

**Reply by fluxoz** [reply]
Oops - I realized I was using gruvbox-material instead, this is currently working for me:

```
    programs.nixvim = {
	enable = true;
        defaultEditor = true;
        extraPlugins = with pkgs.vimPlugins; [
        	gruvbox
        ];
	colorschemes.gruvbox.enable = true;
        plugins.lightline.enable = true;
    };
```

**Reply by MattSturgeon** [reply]
Are you saying that `extraPlugins = [ pkgs.vimPlugins.gruvbox ]` works but `colorschemes.gruvbox.enable = true` does not?

I should note that our `gruvbox` plugin installs `gruvbox-nvim` _not_ `gruvbox`: https://github.com/nix-community/nixvim/blob/5922a48008e5759acb63a12b2de8348ec512760f/plugins/colorschemes/gruvbox.nix#L12

Looking on [nixos search](https://search.nixos.org/packages?channel=unstable&show=vimPlugins.gruvbox&from=0&size=50&buckets=%7B%22package_attr_set%22%3A%5B%22vimPlugins%22%5D%2C%22package_license_set%22%3A%5B%5D%2C%22package_maintainers_set%22%3A%5B%5D%2C%22package_platforms%22%3A%5B%5D%7D&sort=relevance&type=packages&query=gruvbox), they relate to different plugins `pkgs.vimPlugins.gruvbox-nvim` packages https://github.com/ellisonleao/gruvbox.nvim, while `pkgs.vimPlugins.gruvbox` packages https://github.com/morhetz/gruvbox.

It is possible lightline only supports one and not the other.

**Reply by fluxoz** [reply]
Yea that was what I experienced, just trying the configuration in the docs out. I have now moved beyond that since I am porting my entire neovim config to nixvim!
